src/utilities/validateReport.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * Check if the given parameter is a valid Report based
+ * on its data structure and type and return a boolean.
+ *
+ * @param   {object}  parsedData  The data to check.
+ * @param   {string}  type        Type of Report to check.
+ */
+
+import Ajv from 'ajv';
+
+// Expected json structure for Table reports
+const tableReportSchema = {
+  type: 'object',
+  properties: {
+    data: {
+      type: 'object',
+      properties: {
+        header: {
+          type: 'array',
+        },
+        rows: {
+          type: 'array',
+          items: { type: 'array' },
+        },
+        formatters: {
+          type: 'array',
+          items: { type: 'string' },
+        },
+      },
+      required: ['header', 'rows', 'formatters'],
+    },
+  },
+  required: ['data'],
+};
+
+// Expected json structure for BarChart, LineChart or PieChart reports
+const otherReportSchema = {
+  type: 'object',
+  properties: {
+    name: { type: 'string' },
+    data: {
+      type: 'array',
+      items: {
+        type: 'object',
+        properties: {
+          label: { type: 'string' },
+          values: {
+            type: 'array',
+            items: { $ref: '#/definitions/plotValues' },
+          },
+        },
+        required: ['label', 'values'],
+      },
+    },
+  },
+  definitions: {
+    plotValues: {
+      type: 'object',
+      properties: {
+        x: { type: ['string', 'number'] },
+        y: { type: ['string', 'number'] },
+      },
+      required: ['x', 'y'],
+    },
+  },
+  required: ['name', 'data'],
+};
+
+export const validateReport = (parsedData, type) => {
```

Comment:
I'd be tempted to use a structure a little more flexible to changes in the future:

```
const TABLE_SCHEMAS = {
    Table: TableReportSchema,
    Pie: DefaultReportSchema,
    Line: DefaultReportSchema,
    Bar: DefaultReportSchema,
}
```

Then your validate report would be more 

`export const validateReport = (data, type) => return ajv.validate(TABLE_SCHEMAS[type] || {}, data)`

No default case here, so not sure if `ajv.validate({}, data)` would always return false (for when the type does not exist)

Seems a little nicer to me, but it's functionally the same!

Reply:
This is great, I was thinking about something to group all different reports and avoid adding as many cases as reports we have (not a problem with four, but a good practice for longer lists).

Reply:
Done here e7779b1